dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -1308,6 +1311,14 @@ public Workflow manualRegister(@ApiParam(hidden = true) @Auth User user,
                     String toolsJSONTable = lInterface.getContent(w.getWorkflowPath(), getMainDescriptorFile(existingTag).getContent(), extractDescriptorAndSecondaryFiles(existingTag), LanguageHandlerInterface.Type.TOOLS, toolDAO);
                     Set<Image> images = lInterface.getImagesFromRegistry(toolsJSONTable);
                     existingTag.getImages().addAll(images);
+
+                    // Grab checksum for file descriptors if not already available.
+                    for (SourceFile sourceFile : existingTag.getSourceFiles()) {
+                        if (sourceFile.getChecksums() == null || sourceFile.getChecksums().isEmpty()) {
+                            String sha = FileFormatHelper.calcSHA1(sourceFile.getContent());
+                            sourceFile.setChecksums(Collections.singletonList(new Checksum(SHA_TYPE_FOR_SOURCEFILES, sha)));
```

Comment:
This is probably OK, but just checking... Collections.singleton returns an immutable list. If at some point in the future we wanted to add a second checksum in the same session, this could cause problems. But I'm probably overthinking it. I guess if we added a second checksum, we would do it right here.

Reply:
set checksums with arraylist of checksums instead of using singleton